### PR TITLE
Add support for stomp over tls/ssl.

### PR DESCRIPTION
**Pull request by kgoerlitz**
This implements request #6 (Need to support stomp protocol over SSL/TLS connection).

### CODE REVIEW

File: lib/logstash/inputs/stomp.rb [hunk 1]
```diff
@@ -47,9 +53,21 @@ def connect
...
-    @client = OnStomp::Client.new("stomp://#{@host}:#{@port}", :login => @user, :passcode => @password.value)
+    # Set protocol based on wether ssl is true or not
+    if @ssl
+      @protocol = "stomp+ssl"
+      if @verify_ssl
+        @ssl_opts = {}  # default in onstomp is to verify certificates
+      else
+        @ssl_opts = {:verify_mode => OpenSSL::SSL::VERIFY_NONE, :post_connection_check => false}  # disable verification
+      end
+      @client = OnStomp::Client.new("stomp+ssl://#{@host}:#{@port}", :login => @user, :passcode => @password.value, :ssl => @ssl_opts)
```
Comment (jordansissel) [inline lib/logstash/inputs/stomp.rb]: I'm having no luck finding the docs for the `:ssl` setting for OnStomp. Specifically, where does it look for trusted certificate authorities?

If there's no way to configure what certificates to trust, got any ideas on documenting what the default cert trust is?

Comment (kgoerlitz) [inline lib/logstash/inputs/stomp.rb]: I believe that OnStomp, like other SSL/TLS libraries will use the root CAs registered in the system.  The default verification will verify that the server certificate is signed by a root CA and that the certificate is valid for the server host.

Comment (kgoerlitz) [inline lib/logstash/inputs/stomp.rb]: CLA signed.

Comment (jordansissel) [inline lib/logstash/inputs/stomp.rb]: > like other SSL/TLS libraries will use the root CAs registered

Can you confirm that, and if so, update the docs to help users know this - for the new ssl setting you've added?

**Comment by ghost** [comment]
Jenkins standing by to test this. If you aren't a maintainer, you can ignore this comment. Someone with commit access, please review this and clear it for Jenkins to run; then  say 'jenkins, test it'.

**Comment by karmi** [comment]
Hi @kgoerlitz, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in yout Git [commit](https://github.com/logstash-plugins/logstash-input-stomp/pull/7.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

**Comment by kgoerlitz** [comment]
I updated my profile and created a new pull request based on the current
master branch.

**Comment by crashedserver** [comment]
How can one use a self signed certificate?

**Pull request closed without merging**